src/librustc_typeck/check/coercion.rs

```diff
@@ -1119,11 +1127,7 @@ impl<'gcx, 'tcx, 'exprs, E> CoerceMany<'gcx, 'tcx, 'exprs, E>
                 }
             }
             Err(err) => {
-                let (expected, found) = if expression.is_none() {
-                    // In the case where this is a "forced unit", like
-                    // `break`, we want to call the `()` "expected"
-                    // since it is implied by the syntax.
-                    assert!(expression_ty.is_nil());
+                let (expected, found) = if label_expression_as_expected {
                     (expression_ty, self.final_ty.unwrap_or(self.expected_ty))
```

Comment:
This part of the comment was meant to be deleted per https://github.com/rust-lang/rust/issues/41425#issuecomment-296754508. Should I rephrase the remaining part?